controllers/topology/internal/scope/blueprint.go

```diff
@@ -38,6 +40,27 @@ type ClusterBlueprint struct {
 
 	// MachineDeployments holds the MachineDeploymentBlueprints derived from ClusterClass.
 	MachineDeployments map[string]*MachineDeploymentBlueprint
+
+	// machineDeploymentsUpgradeNames captures the list of machine deployments that will be upgraded
+	machineDeploymentsUpgradeNames []string
```

Comment:
I am using a list of machine deployments names instead of counter as we can potentially use this information to log what machine deployments are being upgraded and may be for other tracking reasons?

Also, making this a list of names instead of a counter makes the `TrackMachineDeploymentUpgrade` function signature look cleaner and kind of reads well. 

Open to other suggestions :)